kalite/topic_tools/__init__.py

```diff
@@ -120,14 +122,57 @@ def get_leafed_topics(force=False):
         LEAFED_TOPICS = [topic for topic in get_node_cache()["Topic"].values() if [child for child in topic.get("children", []) if child.get("kind") != "Topic"]]
     return LEAFED_TOPICS
 
+def create_thumbnail_url(thumbnail):
+    if is_content_on_disk(thumbnail, "png"):
+        return settings.CONTENT_URL + thumbnail + ".png"
+    elif is_content_on_disk(thumbnail, "jpg"):
+        return settings.CONTENT_URL + thumbnail + ".jpg"
+    return None
+
 CONTENT          = None
 CACHE_VARS.append("CONTENT")
-def get_content_cache(force=False):
+def get_content_cache(force=False, annotate=False):
     global CONTENT, CONTENT_FILEPATH
 
     if CONTENT is None or force:
         CONTENT = softload_json(CONTENT_FILEPATH, logger=logging.debug, raises=False)
- 
+    
+    if annotate:
+        if settings.HARD_CONTENT_CACHE and not force:
+            content = softload_json(CONTENT_FILEPATH + ".cache", logger=logging.debug, raises=False)
```

Comment:
Haven't followed the whole logic chain through, but will this only get run once per server run, or repeatedly every time the content cache is needed? The latter would seem inefficient, and I guess we would want (and may already have) it running only if the cache isn't already loaded or something?